### PR TITLE
updated install script (`install.sh`)

### PR DESCRIPTION
**Pull request by zackbradys**
**Please check below, if the PR fulfills these requirements:**
- [X] Commit(s) and code follow the repositories guidelines.
- [X] Test(s) have been added or updated to support these change(s).
- [X] Doc(s) have been added or updated to support these change(s).

**Associated Links:**

- https://github.com/hauler-dev/hauler/issues/253

**Types of Changes:**

- Feature

**Proposed Changes:**

- Removed requirement of `sudo` for the install script
- Cleaned up references to variables within the script
- Normalized script and variable syntaxes
- Added the ability to set the installation directory with `HAULER_INSTALL_DIR`
- Added the ability to enable debug mode of the script with `HAULER_DEBUG`

**Verification/Testing of Changes:**

- N/A

**Additional Context:**

- N/A

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -36,27 +40,40 @@ function warn {
...
     echo && echo "[ERROR] Hauler: $1"
-    exit 0
+    exit 1
 }
 
+# exit if unable to run with root privileges
+if [ "$(id -u)" -ne 0 ]; then
+    fatal "Root privileges are required to install Hauler"
+fi
+
```
Comment (dweomer) [inline install.sh]: With the addition of `HAULER_INSTALL_DIR` this check should be conditional on whether or not the target directory (or any missing parents, assuming we create such) is not writable by the calling user.

Comment (zackbradys) [reply install.sh]: Ah good catch... will get this added to this PR

File: install.sh [hunk 2]
```diff
@@ -110,72 +127,64 @@ fi
...
+        source "$HOME/.bash_profile"
+    elif [ -f "$HOME/.zshrc" ]; then
+        echo "export PATH=\$PATH:${HAULER_INSTALL_DIR}" >> "$HOME/.zshrc"
+        source "$HOME/.zshrc"
+    elif [ -f "$HOME/.profile" ]; then
+        echo "export PATH=\$PATH:${HAULER_INSTALL_DIR}" >> "$HOME/.profile"
+        source "$HOME/.profile"
+    else
+        echo "Failed to add ${HAULER_INSTALL_DIR} to PATH: Unsupported Shell"
+    fi
```
Comment (dweomer) [inline install.sh]: If hauler is already in the path, tested via `command -v hauler` and at the same place the we put it (maybe using `realpath` of the result and comparing that to the `realpath` of the installation location) then we should consider skipping this step. @amartin120 what do you think?

**Review by dweomer — DISMISSED** [review]
okay as-is but would prefer some slight changes/considerations

File: install.sh (hunk 2 above)
Comment (dweomer) [reply install.sh]: there is a caveat here in that `realpath` works slightly differently on mac vs linux.

Comment (dweomer) [reply install.sh]: i guess this check on line 165 is probably good enough:
```
if [[ ":$PATH:" != *":${HAULER_INSTALL_DIR}:"* ]]; then
```

Comment (zackbradys) [reply install.sh]: I love overly complex shell expressions in lieu of preexisting simplified linux commands

**Comment by zackbradys** [comment]
```bash
zackbradys@Zacks-MacBook-Pro hauler % curl -sfL https://raw.githubusercontent.com/zackbradys/hauler/main/install.sh | sudo bash

[INFO] Hauler: Starting Preflight Checks...

[INFO] Hauler: Starting Installation...
- Version: v1.0.4
- Platform: darwin
- Architecture: arm64
- Install Directory: /usr/local/bin

[INFO] Hauler: Starting Download...

[INFO] Hauler: Starting Checksum Verification...
- Expected Checksum: b9bd06fdec76bd760a96fe0239c79a3b0a8722417283f12469f66e328340fe13
- Determined Checksum: b9bd06fdec76bd760a96fe0239c79a3b0a8722417283f12469f66e328340fe13
- Successfully Verified Checksum: hauler_1.0.4_darwin_arm64.tar.gz

[INFO] Hauler: Successfully Installed Hauler at /usr/local/bin/hauler

[INFO] Hauler: Hauler v1.0.4 is now available for use!
- Documentation: https://hauler.dev

zackbradys@Zacks-MacBook-Pro hauler % hauler
Airgap Swiss Army Knife

Usage:
  hauler [flags]
  hauler [command]

Available Commands:
  completion  Generates completion scripts for various shells
  help        Help about any command
  login       Log in to a registry
  store       Interact with hauler's embedded content store
  version     Print the current version

Flags:
  -h, --help               help for hauler
  -l, --log-level string    (default "info")

Use "hauler [command] --help" for more information about a command.
```

---

```bash
zackbradys@Zacks-MacBook-Pro hauler % sudo ./install.sh 

[INFO] Hauler: Starting Preflight Checks...

[INFO] Hauler: Starting Installation...
- Version: v1.0.4
- Platform: darwin
- Architecture: arm64
- Install Directory: /usr/local/bin

[INFO] Hauler: Starting Download...

[INFO] Hauler: Starting Checksum Verification...
- Expected Checksum: b9bd06fdec76bd760a96fe0239c79a3b0a8722417283f12469f66e328340fe13
- Determined Checksum: b9bd06fdec76bd760a96fe0239c79a3b0a8722417283f12469f66e328340fe13
- Successfully Verified Checksum: hauler_1.0.4_darwin_arm64.tar.gz

[INFO] Hauler: Successfully Installed Hauler at /usr/local/bin/hauler

[INFO] Hauler: Hauler v1.0.4 is now available for use!
- Documentation: https://hauler.dev

zackbradys@Zacks-MacBook-Pro hauler % hauler
Airgap Swiss Army Knife

Usage:
  hauler [flags]
  hauler [command]

Available Commands:
  completion  Generates completion scripts for various shells
  help        Help about any command
  login       Log in to a registry
  store       Interact with hauler's embedded content store
  version     Print the current version

Flags:
  -h, --help               help for hauler
  -l, --log-level string    (default "info")

Use "hauler [command] --help" for more information about a command.
```

File: install.sh [hunk 3]
```diff
@@ -36,146 +44,174 @@ function warn {
...
+    set -x
+fi
+
+# start hauler preflight checks
+info "Starting Preflight Checks..."
+
+# check for required root privileges
+if [ "$(id -u)" -ne 0 ]; then
+    fatal "Root privileges are required to install Hauler"
+fi
```
Comment (dweomer) [inline install.sh]: this is still premature if HAULER_INSTALL_DIR is writable

File: install.sh [hunk 4]
```diff
@@ -36,146 +44,174 @@ function warn {
...
 function fatal {
     echo && echo "[ERROR] Hauler: $1"
-    exit 0
+    exit 1
 }
 
-# check for required dependencies
-for cmd in sudo rm curl grep mkdir sed awk openssl tar; do
+# debug hauler from argument or environment variable
+if [ "${HAULER_DEBUG}" = "true" ]; then
```
Comment (dweomer) [inline install.sh:51]: we should probably want to use `HAULER_INSTALL_DEBUG` for this as the envvar

File: install.sh [hunk 5]
```diff
@@ -36,146 +44,174 @@ function warn {
...
+# start hauler preflight checks
+info "Starting Preflight Checks..."
+
+# check for required root privileges
+if [ "$(id -u)" -ne 0 ]; then
+    fatal "Root privileges are required to install Hauler"
+fi
+
+# check for required packages and dependencies
+for cmd in echo curl grep sed rm mkdir awk openssl tar; do
```
Comment (dweomer) [inline install.sh]:
```suggestion
for cmd in echo curl grep sed rm mkdir awk openssl tar install; do
```

File: install.sh (hunk 4 above)
Comment (zackbradys) [reply install.sh:51]: I would argue to keep it the same due to the syntax of the other args and vars in the script like `HAULER_UNINSTALL` or `HAULER_VERSION`

**Review by dweomer — APPROVED** [review]

**Pull request merged**